### DISCUSSION TITLE
Proposal to rename pmds to iodev

### DISCUSSION THREAD

**Discussion by maryamtahhan** [CloudNativeDataPlane/cndp#48] [Ideas]
Hi folks
the naming  of the I/O layer as `PMDs` can cause a bit of confusion/heart ache as regards having to explain that there are not actually PMDs in CNDP.  As such, I propose a rename of PMDs to iodev (it keeps inline with other APIs names also: pktdev, xskdev...). This would mean renaming the pmds dir and any documentation that references pmds... 

Would like to hear your thoughts on the subject

**Comment by KeithWiles** [comment]
We would need to remove all references to 'pmd' as well in all of the current pmd files or can we leave them?
I guess we could just rename pmds to iodev and the references to the pmds directory.

As for the name of the directory iodev is good, but I was thinking about 'drvs', but iodev is reasonable.

**Comment by jeffreybshaw** [comment]
I don't mind the drivers being named PMD, but if we think there's some reason to change, I won't object.

Instead of "iodev" maybe it can be "iod" since that has the same number of letters as "pmd" so when we search/replace we won't have too many formatting changes related to lines getting longer. Also, I searched teh internet for "iodev" and there are already some references to it in a computing context, so maybe "IOD" can stand for "I/O Driver".

**Comment by KeithWiles** [comment]
What about iodrv directory and iod for the internal name changes?

**Reply by maryamtahhan** [reply]
I don't mind iod, I just think we should really avoid terminology such as driver or PMD as it has connotations that CNDP has its own user space drivers (rather than leveraging the Linux Kernel drivers and interfaces). PMD specifically as people make the assumption that CNDP is co-dependent on DPDK...

Unfortunately not everybody takes the time to look into the lower level details and simply make deductions from the dir name...

**Comment by maryamtahhan** [comment]
I will take a stab at this rename
